Logger.Tests/BaseLoggerMixinsTests.cs

```diff
@@ -26,14 +26,94 @@ public void Error_WithData_LogsMessage()
             var logger = new TestLogger();
 
             // Act
-            //logger.Error("Message {0}", 42);
+            logger.Error("Message {0}", 42);
 
             // Assert
             Assert.AreEqual(1, logger.LoggedMessages.Count);
             Assert.AreEqual(LogLevel.Error, logger.LoggedMessages[0].LogLevel);
             Assert.AreEqual("Message 42", logger.LoggedMessages[0].Message);
         }
 
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void Warning_WithNullLogger_ThrowsException()
+        {
+            // Arrange
+
+            // Act
+            BaseLoggerMixins.Warning(null!, "");
```

Comment:
```suggestion
            BaseLoggerMixins.Warning(null, "");
```
Mark your `BaseLogger` parameter as nullable.